src/openrct2/world/Banner.cpp

```diff
@@ -441,18 +436,19 @@ sint32 banner_get_closest_ride_index(sint32 x, sint32 y, sint32 z)
         {   0,   0 }
     };
 
-    for (i = 0; i < (sint32)Util::CountOf(NeighbourCheckOrder); i++)
+    for (size_t i = 0; i < (sint32)Util::CountOf(NeighbourCheckOrder); i++)
     {
-        rideIndex = banner_get_ride_index_at(x + NeighbourCheckOrder[i].x, y + NeighbourCheckOrder[i].y, z);
-        if (rideIndex != -1)
+        uint8 rideIndex = banner_get_ride_index_at(x + NeighbourCheckOrder[i].x, y + NeighbourCheckOrder[i].y, z);
+        if (rideIndex != RIDE_ID_NULL)
         {
             return rideIndex;
         }
     }
 
-    rideIndex = -1;
+    sint32 index;
```

Comment:
I believe you can make index an `uint8` here, then you can eliminate the case later. Otherwise it looks good to me.